msal-obo-sample/src/main/java/com/microsoft/azure/msalobosample/OboAuthProvider.java

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.azure.msalobosample;
+
+import com.google.common.hash.Hashing;
+import com.microsoft.aad.msal4j.ClientCredentialFactory;
+import com.microsoft.aad.msal4j.ConfidentialClientApplication;
+import com.microsoft.aad.msal4j.IAuthenticationResult;
+import com.microsoft.aad.msal4j.MsalException;
+import com.microsoft.aad.msal4j.OnBehalfOfParameters;
+import com.microsoft.aad.msal4j.SilentParameters;
+import com.microsoft.aad.msal4j.UserAssertion;
+import com.microsoft.graph.authentication.IAuthenticationProvider;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.cache.CacheManager;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.oauth2.provider.authentication.OAuth2AuthenticationDetails;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.Nonnull;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.concurrent.CompletableFuture;
+
+@Component
+class OboAuthProvider implements IAuthenticationProvider {
```

Comment:
The GraphSDK team wants us to extend [BaseAuthenticationProvider](https://github.com/microsoftgraph/msgraph-sdk-java-core/blob/dev/src/main/java/com/microsoft/graph/authentication/BaseAuthenticationProvider.java) (which in turn implements IAuthenticationProvider). See [here](https://github.com/Azure-Samples/ms-identity-java-spring-tutorial/blob/f654ee9494cd5030fa902c6d5763fd2cc7ebef1e/2-Authorization-I/call-graph/src/main/java/com/microsoft/azuresamples/msal4j/msidentityspringbootwebapp/Utilities.java#L89) for an example.

Reply:
Thanks, its been updated.